dali/pipeline/executor/queue_policy.h

```diff
@@ -285,14 +300,38 @@ struct SeparateQueuePolicy {
     }
   }
 
-  void SignalError() {
-    exec_error_ = true;
+  void NotifyAll() {
     ready_output_cv_.notify_all();
     free_cond_.notify_all();
+    for (int i = 0; i < static_cast<int>(OpType::COUNT); ++i) {
+      stage_ready_cv_[i].notify_all();
+      stage_free_cv_[i].notify_all();
+    }
+  }
+
+  void SignalStop() {
+    {
+      std::lock_guard<std::mutex> lock(ready_output_mutex_);
+      ready_stop_ = true;
+    }
+    for (int i = 0; i < static_cast<int>(OpType::COUNT); ++i) {
+      {
+        std::lock_guard<std::mutex> l(stage_free_mutex_[i]);
+        stage_free_stop_[i] = true;
+      }
+      {
+        std::lock_guard<std::mutex> l(stage_ready_mutex_[i]);
+        stage_ready_stop_[i] = true;
+      }
+    }
+    NotifyAll();
   }
 
-  bool IsErrorSignaled() {
-    return exec_error_;
+  bool IsStopSignaled() {
```

Comment:
Minor perf. issue: this mutex does not guarantee anything more than an atomic variable would. Since the mutex is shared between stages, it seems likely it will actually block.

Reply:
Do you suggest swapping it to an atomic variable?
For the checks should be fine, for the actual stop condition (when used with condtion variable) it would still be propagated under mutex.
@Kh4L what do you think? We should probably check it.

Reply:
Sure @klecki , we can check it

Reply:
Done.